### ISSUE TITLE
Choose License for future repos

### ISSUE THREAD

**Issue by brey** [oceanmodeling/Management#6]
We need to decide which license we should use for new developments. At the JRC we supported the EUPL license which is essentially a multilingual GPL (https://en.wikipedia.org/wiki/European_Union_Public_Licence).

However, the usage of EUPL software assumes that every integration with a GPL component moves everything to GPL (https://www.eupl.eu) so we should just move away from it. 

Personally I would prefer GPLv.3 as the most open and copylefted of the them all. 

What is you opinion?

**Comment by platipodium** [comment]
We have GPL'd www.mossco.de as we're relying on many open source tools, like FABM and GETM (both GPL).  As we're integrating to SCHISM (Apache), we dual-license our development.

We're noticing that the restrictions of GPL an be severe in downstream development when integration with non-GPL is sought

**Comment by josephzhang8** [comment]
SCHISM also uses Apachev2, to allow private collaboration.

**Comment by platipodium** [comment]
There is basically no voting, as it is not in the developer's right to choose, but in the employer's/organization's

**Comment by saeed-moghimi-noaa** [comment]
Good point Carsten!

**Comment by platipodium** [comment]
The inventory of our codes will very likely combine three classes of licenses

- under CC0 or similar (public domain, all US gov't code) (e.g. NEMS, ESMF, ..)
- under Apache or similar (permissive) (e.g. SCHISM)
- under GPL or similar (CeCILL, EUPL) (most open source tools we use, NEMO)

According to this post by Richard Stallmann on combining licenses: https://www.gnu.org/licenses/license-compatibility.html we can be building combined systems out of these codes in a one-way manner

- GPL subsumes Apache
- Apache subsumes CC0

where each of the parts retain their individual licenses.

**For any combined system that uses CC0 and Apache, the combined system's license is then necessarily Apache.
For any combined system that uses GPL and Apache or CC0, the combined license it then necessarily GPL.**

Now as long as we distribute only source code, there is no issue at all with licenses being compatible or incompatible.  But as soon as we distribute libraries, or .. potentially services like OpenCoasts ...we have to be aware of the licensing. 

Showstoppers in all of this are licenses that are not in the public domain/permissive category or have a different copyleft.

TLDR; use CC0-like or Apache-like or GPL-like depending on your employer's strategy and we'll be fine.  Don't choose anything else.

**Comment by brey** [comment]
Thanks to @platipodium primer, this issue can be considered resolved for now.
